app/components/global_threads/thread_item/thread_item.tsx

```diff
@@ -0,0 +1,273 @@
+export type OwnProps = {
+    testID: string;
+    theme: Theme;
+    threadId: string;
+};
+
+export type StateProps = {
+    channel: Channel;
+    post: Post;
+    thread: UserThread | null;
+    threadStarter: UserProfile;
+}
+
+type Props = DispatchProps & OwnProps & StateProps & {
+    intl: typeof intlShape;
+};
+
+function ThreadItem({actions, channel, intl, post, threadId, testID, theme, thread, threadStarter}: Props) {
+    const style = getStyleSheet(theme);
+
+    if (!thread) {
+        return null;
+    }
+
+    let postItem = post;
+
+    React.useEffect(() => {
+        if (!postItem) {
+            // Get the latest post
+            actions.getPost(threadId);
+        }
+    }, []);
+
+    if (!postItem) {
+        // Have the post from thread item
+        postItem = thread.post;
+    }
+
+    const channelName = channel?.display_name;
+    const threadStarterName = displayUsername(threadStarter, Preferences.DISPLAY_PREFER_FULL_NAME);
+
+    const showThread = () => {
+        actions.getPostThread(postItem.id);
+        actions.selectPost(postItem.id);
+        const passProps = {
+            channelId: postItem.channel_id,
+            rootId: postItem.id,
+        };
+        goToScreen(THREAD, '', passProps);
+    };
+
+    const testIDPrefix = `${testID}.${postItem?.id}`;
+
+    const needBadge = thread.unread_mentions || thread.unread_replies;
+    let badgeComponent;
+    if (needBadge) {
+        if (thread.unread_mentions && thread.unread_mentions > 0) {
+            badgeComponent = (
+                <View
+                    style={style.mentionBadge}
+                    testID={`${testIDPrefix}.unread_mentions`}
+                >
+                    <Text style={style.mentionBadgeText}>{thread.unread_mentions > 99 ? '99+' : thread.unread_mentions}</Text>
+                </View>
+            );
+        } else if (thread.unread_replies && thread.unread_replies > 0) {
+            badgeComponent = (
+                <View
+                    style={style.unreadDot}
+                    testID={`${testIDPrefix}.unread_dot`}
+                />
+            );
+        }
+    }
+
+    let name;
+    if (postItem.state === Posts.POST_DELETED) {
+        name = (
+            <Text
+                style={[style.threadStarter, style.threadDeleted]}
+                numberOfLines={1}
+            >{intl.formatMessage({
+                    id: 'threads.deleted',
+                    defaultMessage: 'Original Message Deleted',
+                })}</Text>
+        );
+    } else {
+        name = (
+            <Text
+                style={style.threadStarter}
+                numberOfLines={1}
+            >{threadStarterName}</Text>
+        );
+    }
+
+    let postBody;
+    if (postItem.state !== Posts.POST_DELETED) {
+        postBody = (
+            <Text
+                style={style.message}
+                numberOfLines={2}
+            >
+                <RemoveMarkdown value={postItem.message || ''}/>
```

Comment:
We are limiting the number of lines to 2 and rendering markdown might make the height as variable which should be then controlled with `overflow` `hidden`.

Reply:
But then this means that I would see a message like 
```
### this is a header
`some text` and a [url](https://someurl.com)
```

@matthewbirtch is that expected?

Reply:
It will appear like this .. Reusing the same component which removes markdown like Pinned messages in channel header.
<img width="376" alt="Screenshot 2021-07-20 at 13 42 09" src="https://user-images.githubusercontent.com/3680799/126285651-791f0e21-73bc-4326-9d4c-e1a7bbe31398.png">